Makefile

```diff
@@ -64,7 +64,7 @@ IMG_SHA = $(IMAGE_TAG_BASE):$(BUILD_SHA)
 CRD_OPTIONS ?= "crd:trivialVersions=true,preserveUnknownFields=false"
 # ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
 ENVTEST_K8S_VERSION = 1.23
-GOLANGCI_LINT_VERSION = v1.42.1
+GOLANGCI_LINT_VERSION = v1.50.1
```

Comment:
I had to update Golint because it complaints with the vendored dependencies, which are in a newer format:
```
controllers/reconcilers/client_helper.go:13:2: could not import k8s.io/apimachinery/pkg/types (-: could not load export data: cannot import "k8s.io/apimachinery/pkg/types" (unknown iexport format version 2), export data is newer version - update tool) (typecheck)
```
We didn't realize it because it seems we are not running linting at any stage of the CI.